src/components/Inputs/FieldSelector.tsx

```diff
@@ -154,6 +161,12 @@ const FieldSelector = forwardRef<HTMLDivElement, FieldSelectorType>(
       setSearchValue(event.target.value);
     };
 
+    useOutsideClick(menuRef.current, (e) => {
+      if (e.target !== buttonRef.current) {
```

Comment:
Wouldnt this close the menu when just clicking inside it? (not on a button inside it but just white-space)